helpers/content.go

```diff
@@ -335,6 +335,15 @@ func ExtractTOC(content []byte) (newcontent []byte, toc []byte) {
 
 	newcontent = append(content[:startOfTOC], content[endOfTOC:]...)
 	toc = append(replacement, origContent[startOfTOC+len(first):endOfTOC]...)
+
+	emptyTop := []byte(`<ul>
+<li>
+`)
+	emptyBottom := []byte(`</ul></li>
+`)
+
+	toc = []byte(strings.Replace(string(toc), string(emptyTop), "", 10))
+	toc = []byte(strings.Replace(string(toc), string(emptyBottom), "", 10))
```

Comment:
* There is a `bytes.Replace` ... making this simpler.
* If you use a variable once, you might as well get rid of it (empty*)
* The last argument should be 1 (one replacement, if there are more there is something weird happening)